### ISSUE TITLE
rpc: `listunspent` doesn't return immature coinbase outputs

### ISSUE THREAD

**Issue by danielabrozzoni** [bitcoin/bitcoin#25728]
**Is your feature request related to a problem? Please describe.**

I need to know all the utxos I have in my wallet, including the immature ones. When using listunspent, the immature coinbases are not returned:
```  
❯ bitcoin-cli listtransactions
[
  {
    "address": "bcrt1q0epv47eaxhyenhqcjp8v7wytjx26k35r6w2py6",
    "category": "immature",
    "amount": 50.00000000,
    "label": "",
    "vout": 0,
    "confirmations": 1,
    "generated": true,
    "blockhash": "334996db387320711e1176f2591dfc16fe8d92f617af47ae65b97c2a817758b0",
    "blockheight": 102,
    "blockindex": 0,
    "blocktime": 1658995253,
    "txid": "480f0340a98755cfb095743b5464e87d0cf1b0200d204507ec8448962bcafbe6",
    "walletconflicts": [
    ],
    "time": 1658995253,
    "timereceived": 1658995253,
    "bip125-replaceable": "no"
  }
]
❯ bitcoin-cli listunspent
[
]
```

**Describe the solution you'd like**

What do you think of adding a `include_immature_coinbase` (default to false) parameter to `listunspent`? If you think it'd be useful, I can start working on this myself.

**Describe alternatives you've considered**

At the moment I'm calling `listtransaction` and adding to the utxo list every immature output I see (https://github.com/bitcoindevkit/bdk/pull/687). This works just fine, but having the `include_immature` parameter would make the code a bit cleaner (and it would save one RPC call!).

**Comment by evanlinjin** [comment]
+1 for an `include_immature_coinbase` parameter.

**Comment by furszy** [comment]
quickly drafted it in #25730. We could probably include them under the `include_unsafe` flag response as well (open for discussion there).

**Comment by evanlinjin** [comment]
> quickly drafted it in #25730. We could probably include them under the `include_unsafe` flag response as well (open for discussion there).

Thank you for the prompt PR @furszy ! An `include_unsafe` flag will be useful as well.

I will try give a proper review of the PR later on, but I had a skim over it and I'm wondering whether it would be easy to update the `RPCResult` to show if the output is an immature coinbase?

**Comment by furszy** [comment]
it would be easy to add but I'm a ~0 there. Mainly because such information can be obtained, if needed, using another command like `gettransaction <txid>`.
(The initial filter makes sense because we are by default, and without a way to change it, not returning immature UTXOs in `listunspent`)

**Comment by achow101** [comment]
> quickly drafted it in #25730. We could probably include them under the `include_unsafe` flag response as well (open for discussion there).

No, "unsafe" has a specific meaning in the wallet which does not include immature coinbases. Unsafe means that the UTXO could disappear as it is created by an untrusted party and the transaction creating it is still unconfirmed.

An immature coinbase is not unsafe. It just makes the transaction temporarily invalid. We shouldn't combine those two definitions.

**Comment by evanlinjin** [comment]
> it would be easy to add but I'm a ~0 there. Mainly because such information can be obtained, if needed, using another command like `gettransaction <txid>`. (The initial filter makes sense because we are by default, and without a way to change it, not returning immature UTXOs in `listunspent`)

If that is the case, I feel like a separate command such as `listimmatureunspent` would be more useful.

**Comment by furszy** [comment]
> > it would be easy to add but I'm a ~0 there. Mainly because such information can be obtained, if needed, using another command like `gettransaction <txid>`. (The initial filter makes sense because we are by default, and without a way to change it, not returning immature UTXOs in `listunspent`)
> 
> If that is the case, I feel like a separate command such as `listimmatureunspent` would be more useful.

It's more probable to get consensus over adding an `is_coinbase` field to the `listunspent` RPC response than creating a new RPC command just for it.

**Comment by kouloumos** [comment]
Note: The `include_immature_coinbase` flag has been added to the `*receivedby*` RPCs with https://github.com/bitcoin/bitcoin/pull/14707

**Comment by aureleoules** [comment]
This issue can be closed now that #25730 is merged.
